pyproject.toml

```diff
@@ -71,49 +71,161 @@ line-length = 79
 indent-width = 4
 
 target-version = "py38"
+exclude = [
+  "CSV",
+  "ref.*?py",
+]
 
 [tool.ruff.lint]
-fixable = ["ALL"]
-preview = true
+fixable = []
+# preview = true
+# logger-objects = []
+task-tags = ["TODO", "FIXME", "XXX"]
 select = [
+  "A", # flake8-builtins
+  "ARG", # flake8-unused-arguments
+  "COM", # flake8-commas
+  "D", # pydocstyle
   "E", # pycodestyle
+  "E701", # multiple-statements-on-one-line-colon
+  "E722", # bare-except
   "F", # Pyflakes
+  "F401", # unused-import
+  "G", # flake8-logging-format
+  "I", # isort
+  "N", # pep8-naming
+  "NPY", # numpy-specific
+  "PL", # Pylint
+  "Q", # flake8-quotes
+  "UP", # pyupgrade
   "W", # Warning
   ]
 
 ignore = [
-  "B008", # function-call-in-default-argument
-  "B024", # abstract-base-class-without-abstract-method
-  "B028", # no-explicit-stacklevel
-  "B905", # zip-without-explicit-strict
+  # Currently don't conform but we might want to reconsider
+  "A001", # builtin-variable-shadowing
+  # Currently don't conform but we might want to reconsider
+  "A002", # builtin-argument-shadowing
+  "ARG002", # unused-method-argument
```

Comment:
I think we should keep all but ARG005 as default, and require line specific overrides. I think this is basically what weve been doing for pylint

Reply:
I think it is a good idea, but as it is, if I don't ignore those, it gives errors.

Reply:
Yeah well have to add inline ruff overrides along side the pylint ones

Reply:
As per our discussion, well remove these ignores in the next PR

Reply:
I'm closing this 'conversation'. Please, feel free to re-open if we're missing anything.